wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -187,7 +227,179 @@ export default {
           }
         }
       })(_this);
-    }
+    },
+
+    /**
+     * This function gets executed after fetching posts data to set the Posts array.
+     * We are initializing the pagination array.
+     **/ 
+
+    process: function(data, query, headers) {
+      // If there are posts for this query, map them to the template.
+      let posts = (Array.isArray(data)) ?
+        data.map(this.maps()[this.type]) : false;
+
+      /**
+       * Set only current page posts
+       * The query argument gets the current, previous, next page
+       * So, we set Posts array with current page posts
+       */
+      
+      if(this.query.page == query.page){
+        this.$set(this.posts[query.page], 'posts', posts);
+        this.$set(this.posts[query.page], 'headers', Object.freeze(headers));
+      }
+
+      // If there are no posts, pass along to the error handler.
+      if (!Array.isArray(data))
+        this.error({error: data, query: query});
+
+      this.$set(this, 'init', true);
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+        this.totalPages = [2,3,4,5]
+      } 
+    },
+
+    /**
+     * Fetch the posts of updated page
+     */
+    updatePagination: function(){
+      this.scrollToTop();
+      let url = [
+        this.domain,
+        this.lang.path,
+        this.endpoints[this.type],
+        this.buildUrlQuery(this.query)
+      ].join('');
+
+      fetch(url)
+      .then(this.response)
+      .then(data => {
+        let headers = Object.assign({}, this.headers);
+        this.posts.length=1;
+        let currentPost = {
+          'posts': data,
+          'headers': headers,
+          'query': this.query,
+          'show': true
+        };
+        this.posts.push(currentPost);
+        this.setPagination();      
+      }).catch(this.error);
+    },
+
+    previousPage:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage-1);
+      this.updatePagination();
+    },
+
+    nextPagination:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage+1);
+      this.updatePagination();
+    },
+
+    immediatePage: function(event){
+      let change = parseInt(event.target.dataset.amount);
+      this.$set(this.query, 'page', change);
+      this.updatePagination();
+    },
+
+    updateQuery: function(taxonomy, terms) {
```

Comment:
Can you add a comment on why it is necessary to override updateQuery?

Reply:
The existing updateQuery method updates 'show' property in the posts array to hide all posts. We are overriding the method to remove all posts except the current page posts.

Reply:
We no need to do this Override, I will delete this section.